### PR TITLE
PA transpose for MixedVectorCurlIntegrator

### PR DESCRIPTION
**Pull request by dylan-copeland**
- [x] Expanded the unit tests (some were excluded by a previous PR).
- [x] Fixed a bug in `PAHcurlHdivSetup2D` revealed by the unit tests.
- [x] Implemented `AddMultTransposePA` for `VectorFEMassIntegrator`.
- [x] Implemented `AddMultTransposePA` for `MixedVectorCurlIntegrator`.
- [x] Implemented `AddMultTransposePA` for `MixedVectorWeakCurlIntegrator` for H(curl) x H(div).
- [x] Tested with CUDA.

Note that some of these integrators do not have shared memory versions, so they will run out of memory quickly (with respect to polynomial order), especially in 3D.  

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2921](https://github.com/mfem/mfem/pull/2921) | @dylan-copeland | @tzanio | @mlstowell + @v-dobrev + @YohannDudouit + @tzanio | 3/24/22 | 4/11/22 | 4/15/22 | |

PR Checklist
    
- [x] Code builds.
- [x] Code passes `make style`.

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by pmocz — APPROVED** [review]
I was able to test this on a GPU and it looks like it is working

**Comment by tzanio** [comment]
@v-dobrev, @YohannDudouit and @mlstowell -- can I get another review on this PR?

File: fem/bilininteg_vectorfe.cpp [hunk 1]
```diff
@@ -372,8 +372,8 @@ void PAHcurlHdivSetup2D(const int Q1D,
                const double R21 = D2*J21;
                const double R22 = D2*J22;
                y(i11,qx,qy,e) = w_detJ * ( J22*R11 - J12*R21); // 1,1
-               y(i12,qx,qy,e) = w_detJ * ( J22*R12 - J12*R22); // 1,2
-               y(i21,qx,qy,e) = w_detJ * (-J21*R11 + J11*R21); // 2,1
+               y(i21,qx,qy,e) = w_detJ * ( J22*R12 - J12*R22); // 1,2 (transpose)
+               y(i12,qx,qy,e) = w_detJ * (-J21*R11 + J11*R21); // 2,1 (transpose)
```
Comment (YohannDudouit) [inline fem/bilininteg_vectorfe.cpp:376]: Was this a bug?

**Review by YohannDudouit — APPROVED** [review]
Thanks @dylan-copeland !

File: fem/bilininteg_vectorfe.cpp (hunk 1 above)
Comment (dylan-copeland) [reply fem/bilininteg_vectorfe.cpp:376]: Yes, I fixed multiple bugs like this in a previous PR. The unit tests used to be on meshes aligned with the coordinate axes, in which case the bug had no effect. I made the unit tests use more general meshes to expose bugs like this.

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by tzanio** [comment]
Do you understand this error from the merge in `next`?

https://github.com/mfem/mfem/runs/5981218867?check_suite_focus=true

**Comment by dylan-copeland** [comment]
> Do you understand this error from the merge in `next`?
> 
> https://github.com/mfem/mfem/runs/5981218867?check_suite_focus=true

I need to merge master into this branch, due to https://github.com/mfem/mfem/pull/2836.

**Comment by tzanio** [comment]
ok, I'll take care of it

**Comment by tzanio** [comment]
Re-merged in `next` for testing...

**Comment by dylan-copeland** [comment]
@tzanio in order for this branch to build after pulling from master, the lines bilininteg_vectorfe.cpp:1094 and 1102 just need to have `|| SMQ` deleted.

**Pull request merged**